src/FFmpegWriter.cpp

```diff
@@ -457,6 +459,65 @@ void FFmpegWriter::SetOption(StreamType stream, string name, string value) {
 				}
 			}
 #endif
+		}
+		// Color range, primiries, transfer characteristics, colorspace output settings.
+		// It also sets file container flags for color details if supported.
+		//
+		// The option may be set as keyword like:
+		//   SetOption(openshot.VIDEO_STREAM, "colorspace", "bt709")
+		//
+		// or integer number of the required numeration like:
+		//   SetOption(openshot.VIDEO_STREAM, "colorspace", str(int(1)) )
+		//
+		// the first one for convenience, the later - for future use.
+		//
+		// To preserve current OpenShot behavior and for compatibility with old hardware and software,
+		// do not set colorspace details if BT.601 standard is in use.
+		// The default color details of ITU-R BT601-6 625, ITU-R BT1358 625, ITU-R BT1700 625 PAL & SECAM, IEC 61966-2-4 xvYCC601
+		// will be applied in this case without setting file container flags (mainly used for SD quality videos).
+#if LIBAVFORMAT_VERSION_MAJOR >= 53
+		else if (c->codec_type == AVMEDIA_TYPE_VIDEO) {
```

Comment:
codec_type should be `AV_GET_CODEC_TYPE(av_stream)`, again for the same reasons... compatibility with different versions of FFmpeg.

Reply:
And again, I don't understand you. The `c->` already points to codec context here, so it has `codec_type` field. For what reason I should re-use the stream's pointer to the same structure? The field itself available in all FFmpeg versions of AVCodecContext.

Enumeration may has different name - but I can use just `0` to make simple and robust comparison (and leave enumeration's name as comment in the code). Is this solution better? No excessive `ifdef` constructions. Usually, when enumeration changes - new values added to the end, so video always will be 0.

Reply:
`av_stream->codecpar->codec_type`  vs  
`av_stream->codec->codec_type`

For some reason, LibAV/FFmpeg devs love to refactor things and rename things. Which path to use is based in FFmpegUtilities.h, which we only built to get past certain "version-based" differences such as this. Hope that makes sense.

Reply:
`c` - is pointer to AVCodecContext structure
declaration:
https://github.com/OpenShot/libopenshot/blob/cf9fbf4402e4a154c741c383b24b6df306e27cd7/src/FFmpegWriter.cpp#L327
definition:
https://github.com/OpenShot/libopenshot/blob/cf9fbf4402e4a154c741c383b24b6df306e27cd7/src/FFmpegWriter.cpp#L334
It's a few lines before my code. All structure entries I use is always part of the AVCodecContext.

For what reason you are create the AV_GET_CODEC_TYPE and why should I use it **exactly** here, if I'm already working with the codec context? Because it is still requires to check if the:

`codec_type` is part of the `enum CodecType` (https://ffmpeg.org/doxygen/0.5/structAVCodecContext.html) or
`codec_type` is part of the `enum AVMediaType` (https://ffmpeg.org/doxygen/1.0/structAVCodecContext.html)

while the AV_GET_CODEC_TYPE still uses same codec context pointer (`av_stream->codec->...`) for all LIBAVFORMAT_VERSION_MAJOR = 55 and below.

If you think that using the AV_GET_CODEC_TYPE is robust solution - then it is not (just different place of the parameter). And sure, I think the `#if LIBAVFORMAT_VERSION_MAJOR >= 53` of mine should be removed to not confuse anybody. I'm start thinking that hard coded zero is the best solution here.